### PR TITLE
adding docs about using `pyproject.toml`

### PR DESCRIPTION
**Pull request by Borda**
Adding note about using `pyproject.toml` which was added recently:
ref:
- https://github.com/crate-ci/typos/pull/790#issuecomment-1735033152
- https://github.com/crate-ci/typos/issues/361#issuecomment-1712008282

### CODE REVIEW

**Comment by Borda** [comment]
@epage mind have look and approve CI run, pls :rabbit:

File: README.md [hunk 1]
```diff
@@ -103,6 +103,22 @@ If you need some more flexibility, you can completely exclude some files from co
...
+extend-ignore-identifiers-re = ["AttributeID.*Supress.*"]
+
+[tool.typos.default.extend-identifiers]
+AttributeIDSupressMenu = "AttributeIDSupressMenu"
+
+[tool.typos.default.extend-words]
+teh = "teh"
+```
+
 ### Integrations
```
Comment (epage) [inline README.md:122]: FYI this was covered in #826.  If we want to expand it, we should probably do it in that area within the context of whats there

Comment (Borda) [reply README.md:122]: I may suggest to have it readme as it is easy to find and the other location was not discoverable =( so I found it based on issue and merged PR...

Comment (epage) [reply README.md:122]: The README is focused on being a quick start, delegating to this file for the details.

Comment (Borda) [reply README.md:122]: Yes but for some cases like us, the need to add a new config in the root could be a deal breaker so in such cases we cannot use it...

File: README.md [hunk 2]
```diff
@@ -103,6 +103,22 @@ If you need some more flexibility, you can completely exclude some files from co
...
+```toml
+[tool.typos.default]
+extend-ignore-identifiers-re = ["AttributeID.*Supress.*"]
+
+[tool.typos.default.extend-identifiers]
+AttributeIDSupressMenu = "AttributeIDSupressMenu"
+
+[tool.typos.default.extend-words]
+teh = "teh"
+```
```
Comment (Borda) [inline README.md:120]:
```suggestion
In addition, you can set the same configuration with standard python's config file `pyproject.toml` by adding `[tool.typos.xyz]` where the `xyz` is the same as in default `_typos.toml`.
```
make it lighter

**Comment by epage** [comment]
This is not aligning with the goal of this section of the documentation.  This is a getting started and isn't meant to cover every feature people might need to use.

**Comment by Borda** [comment]
> This is not aligning with the goal of this section of the documentation. This is a getting started and isn't meant to cover every feature people might need to use.

do you call configuration a feature? :)

File: README.md (hunk 1 above)
Comment (mecampbellsoup) [reply README.md:122]: also the docs added in #826 are crappy and insufficient. in general this project suffers from not having enough examples to get up the learning curve _immediately_

**Pull request closed without merging**